### ISSUE TITLE
Incorrect token type given to some resource attributes

### ISSUE THREAD

**Issue by alexjfisher** [puppetlabs/puppet-lint#232]
If the name of a resource attribute clashes with a Puppet keyword, then puppet-lint does not assign it the `:NAME` token type, but a keyword token type instead.

For example, with this `Exec` resource

```puppet
exec { '/bin/some_command':
  unless => '/bin/foo',
}
```

The `unless` bare word is assigned the `:UNLESS` token type.  Since in this context, it is not the `unless` puppet keyword, it should be assigned the `:NAME` token type instead.

This also causes issues when using `resource_indexes`. The `:param_tokens` hash element for the above example is an empty array, when it should contain a single token (for the single attribute used).

From http://puppet-lint.com/developer/api/
> :param_tokens - An Array of [PuppetLint::Lexer::Token](http://puppet-lint.com/developer/tokens/) objects for the parameter names in the resource declaration.
